src/tools/cephfs_mirror/PeerReplayer.cc

```diff
@@ -2105,6 +2105,13 @@ void PeerReplayer::run(SnapshotReplayerThread *replayer) {
 
       last_directory_scan = now;
     }
+    for (auto &[dir_root, sync_stat] : m_snap_sync_stats) {
+      if (sync_stat.failed) {
+	std::vector<DaemonHealthMetric> health_metrics;
+	health_metrics.emplace_back(daemon_metric::MIRRORING_FAILURE, 1, ceph_clock_now());
```

Comment:
So, with the same health metric type (`MIRRORING_FAILURE`) the mirror daemon is sending the number of file systems that have failed mirroring and/or the number of the dirs that have mirror failures. How is that supposed to show up in the health warning message?

Reply:
We need health warnings in the following cases
- cephfs-mirror fails to enable mirroring on a file system
-  cephfs-mirror hits the upper limit of consecutive failures when synchronizing snapshots for a directory

It would be nice to have different warning messages for the above. We should start with this. Later we can add warnings to issues like slow sync of snapshots and other miscellaneous issues.